pkg/lib/accountmanagement/service.go

```diff
@@ -101,24 +288,369 @@ type AuthenticationInfoService interface {
 	Save(entry *authenticationinfo.Entry) error
 }
 
+type PasskeyService interface {
+	ConsumeAttestationResponse(attestationResponse []byte) (err error)
+}
+
 type SettingsDeleteAccountSuccessUIInfoResolver interface {
 	SetAuthenticationInfoInQuery(redirectURI string, e *authenticationinfo.Entry) string
 }
 
-type UserService interface {
-	UpdateMFAEnrollment(userID string, t *time.Time) error
+type OTPSender interface {
+	Prepare(channel model.AuthenticatorOOBChannel, target string, form otp.Form, typ otp.MessageType) (*otp.PreparedMessage, error)
+	Send(msg *otp.PreparedMessage, opts otp.SendOptions) error
+}
+
+type OTPCodeService interface {
+	GenerateOTP(kind otp.Kind, target string, form otp.Form, opt *otp.GenerateOptions) (string, error)
+	VerifyOTP(kind otp.Kind, target string, otp string, opts *otp.VerifyOptions) error
+}
+
+type VerificationService interface {
+	NewVerifiedClaim(userID string, claimName string, claimValue string) *verification.Claim
+	MarkClaimVerified(claim *verification.Claim) error
+	GetIdentityVerificationStatus(i *identity.Info) ([]verification.ClaimStatus, error)
 }
 
 type Service struct {
 	Database                  *appdb.Handle
+	Config                    *config.AppConfig
+	Challenges                ChallengeProvider
+	Users                     UserService
 	Store                     Store
 	OAuthProvider             OAuthProvider
+	BiometricProvider         BiometricIdentityProvider
 	Identities                IdentityService
 	Events                    EventService
 	Authenticators            AuthenticatorService
 	AuthenticationInfoService AuthenticationInfoService
+	PasskeyService            PasskeyService
 	UIInfoResolver            SettingsDeleteAccountSuccessUIInfoResolver
-	Users                     UserService
+	OTPSender                 OTPSender
+	OTPCodeService            OTPCodeService
+	Verification              VerificationService
+}
```

Comment:
Let's introduce some facade classes for identity related operations.
We should keep token related operations in Service.

```go
func (s *Service) AddPhoneIdentity(input ResumeAddPhoneIdentityInput) error {
  result := s.IdentityFacade.AddPhoneIdentity(token)
  token := s.Store.GenerateToken(....)
  return 
}

func (s *Service) ResumeAddPhoneIdentity(input ResumeAddPhoneIdentityInput) error {
  token, err := s.Store.GetToken(input.Token)
  .....
  s.IdentityFacade.ResumeAddPhoneIdentity(token)
}


```

Reply:
1. Create an identityInfo
2. Check duplication
3. Check if needed to verify
    1. Send code if needed
4. Insert to DB
5. Dispatch Identity created event
```go

func (f *Facade) CreateIdentity(input...) error {
  identity := f.Identity.New()

  // check duplication

  if needVerify {
     // generate code
     // send code
     return ....
  }

  err := f.Identity.Create(identity)
  e := 
}


func (f *Facade) VerifyCreateIdentity(input...) error {
  // check code
  //
  f.CreateIdentity()
  e := VerifyEvent
  f.Event.Dispatch(e)
}
```